### ISSUE TITLE
Image randomly disappear when I put image in the header and multiple images in the body

### ISSUE THREAD

**Issue by cw0516** [elapouya/python-docx-template#372]
## Describe the bug
When I put static image or Inline variable image in the header and then put multiple images in the body,
an image of the list of multiple images disappear randomly

#### here is my github where runnable test case exist

https://github.com/cw0516/docxtplBug

#### and here is same python code in my githhub

////////////////////////////////////////////////////////////////////
from docxtpl import DocxTemplate, InlineImage
from docx.shared import Mm

tpl = DocxTemplate('templates/header_footer_inline_image_tpl.docx')

context = {
    'inline_image': InlineImage(tpl, 'templates/django.png', height=Mm(10)),
    'images': [
        InlineImage(tpl, 'templates/django.png', height=Mm(10)),
        InlineImage(tpl, 'templates/django.png', height=Mm(10)),
        InlineImage(tpl, 'templates/django.png', height=Mm(10))
    ]
}
tpl.render(context)
tpl.save('output/header_footer_inline_image.docx')
///////////////////////////////////////////////////////////////////

#### here is same template docx and image in my github 
![django](https://user-images.githubusercontent.com/48001039/126587362-92328ab8-d7ef-4f68-bc87-b6fcf4e8fff0.png)
[header_footer_inline_image_tpl.docx](https://github.com/elapouya/python-docx-template/files/6859853/header_footer_inline_image_tpl.docx)

## Expected behavior
An image in the header and images in the body should be printed normally

## Screenshots

![screenshot](https://user-images.githubusercontent.com/48001039/126587050-ea44cfb0-9840-4c7c-b075-a6b9d86e6d63.JPG)

**Comment by elapouya** [comment]
Cannot reproduce : Tested on Office365 online : generated docx is displayed correctly, even by updating, saving + re-opening. Tried many times, tried with more inline images in body too.
Was running docxtpl 0.11.5 and python 3.9

**Comment by cw0516** [comment]
How many images printed?? In my python code, 3 images should be printed but it shows only 2

@srutherfordbta how about you??

**Comment by elapouya** [comment]
All images are printed : 1 in the Header, 3 in the body. I added up to 7 images in the body and it always worked as expected.

**Comment by srutherfordbta** [comment]
@cw0516 I executed your code on my laptop got the same results with you. Only two of the three images showing in the body.
Word - Version 2106 (Build 14131.20332 Click-to-Run)
Windows Windows 10 Pro (Version 10.0.19041 Build 10941)
Python 3.7.9 64-bit
docxtpl 0.11.5

![image](https://user-images.githubusercontent.com/87780853/127661347-c9762775-9188-403a-b8ad-55faf88474bc.png)

**Comment by srutherfordbta** [comment]
I uploaded the output file to office 365 and viewed the document. It rendered successfully. I think what we have is a rendering bug in the Windows version of MS-Word. @cw0516 can you provide your Windows and Word builds for the record?
![image](https://user-images.githubusercontent.com/87780853/127663944-56c48031-61a0-4802-b6c7-24611d2d6d40.png)

**Comment by srutherfordbta** [comment]
I forced an MS Office update and Word updated to Version 2107 (Build 14228.20204 Click-to-Run). The rendering issue is still there.

**Comment by cw0516** [comment]
yeah my word version on this issue was 2107(build 14228.20204) and my os is window 10 pro

**Comment by srutherfordbta** [comment]
@elapouya I opened the output in Office365 which renders correctly and added two spaces below the third django image in the body. I then saved it to my local laptop. I opened it in Windows Word and it is just fine. Would you mind comparing the internals of this document to the one on @cw0516 git repo output folder to see what the difference is short of the two spaces? I hope it is something actionable so we can workaround the rendering issue in Windows Word when using docxtpl to create documents.

[header_footer_inline_image.docx](https://github.com/elapouya/python-docx-template/files/6908228/header_footer_inline_image.docx)

**Comment by elapouya** [comment]
Made a diff between before and after modifying on Office365 : there are some differences, but I cannot see something making images hidden randomly.

<img width="1440" alt="snapshot_docx_office365" src="https://user-images.githubusercontent.com/888300/127689612-fcd0b1c6-a025-4d0f-b655-21b1f334b3df.png">

The problem is that this xml code is generated with python-docx and I cannot see what to do with the problem.

**Comment by srutherfordbta** [comment]
Would you mind creating an issue on the python-docx git explaining what your code does and what you are seeing? Maybe they have some ideas or hopefully a fix.
https://github.com/python-openxml/python-docx/issues

**Comment by elapouya** [comment]
That is not that easy : InlineImage has been modified by many people on this project, using internals of python-docx that I do not know. The first thing I have to do is to have a test case. That means I can reproduce the problem (which is not my case) and isolate the faulty xml code. Right now I do not own MS Word at your version, I have to buy one ...

**Comment by srutherfordbta** [comment]
You can get an office 365 account at $7 a month which includes Word for both PC and Mac. If you got a paypal account or  whatever I would be happy to help out. You have done such a great job with docx-tpl and remained active with new features and maintenance.

**Comment by elapouya** [comment]
Thank you a lot for your proposal :) But I can buy one myself. Unfortunately my internet access has actually some troubles, I cannot download big files. I expect to have a solution within 2 weeks.

**Comment by elapouya** [comment]
While coding another feature, I discovered that templates themselves could be "corrupted" in some way or at least not understood totally by python-docx. Mines was created a long time ago with MS Word 1997. 
Could you do 2 tests for me : 
With https://github.com/cw0516/docxtplBug test case, could you : 
1) Try to "sanitize" templates/header_footer_inline_image_tpl.docx by reading it by office365, saving it again with the same name and run the test again. 
2) Instead of giving 3 times the same image, could you try with 3 different images ?

**Comment by cw0516** [comment]
https://user-images.githubusercontent.com/48001039/127737839-40001d25-e6fb-4e35-aa4e-c11a9a7a44b7.mp4

I tried first test. Is this right?

**Comment by elapouya** [comment]
Yes, thank you, it was what I wanted : obviously, the problem remains. Could you try 2nd test ?

**Comment by cw0516** [comment]
sure here is code , images and result

![image](https://user-images.githubusercontent.com/48001039/127743287-81c85727-ee75-4ea8-b14b-155187f9f65d.png)

**Comment by elapouya** [comment]
Thank you a lot, this is really incredible.
Could you swap line 11 and 12 to see whether it is always the first image ?

**Comment by cw0516** [comment]
yeah It's hard for me to know what happen internally ..

![image](https://user-images.githubusercontent.com/48001039/127743736-dd04374e-644a-43d1-8cf6-3e0924551dd6.png)

**Comment by elapouya** [comment]
With what you've given to me, I think it is a matter of .docx "relations ID"  : one is missing or there is a shift in IDs
My little finger tells me that if you remove totally the header, the 3 images will be displayed.
I will look again the internal of the .docx in this direction...

**Comment by cw0516** [comment]
yeah In my project, There was no problem of printing images in body but after I inserted image in the header, this issue happend. Anyway thank you for your work and  I will keep watching this issue as a lover of docxtpl😄

**Comment by elapouya** [comment]
I compared docx internals between your generated docx and the same but modified on office 365 online : I cannot see obvious differences : all internal .xml files are valid (tested in a XML checker). Relationships ID has been re-numbered but still pointing on the same resources. The things I noticed : 'eastAsia' related tags has been removed with office 365 online, `xmlns` attributes are not always on the same tag. 
XML syntax do not differ between image1, image2 and image3 in the body, so if first is not seen the others should not be seen too which is not the case.

I really need the MS Word version you are using to reproduce my side and do deeper testings, I will be able to get that version within 2 weeks I think...

**Comment by dragosnsandu** [comment]
Hi all,

I am currently experiencing the same issue described here. We are using _python-docx_ to generate a DOCX file and _python-docx-template_ to overwrite template tags with some specific content (part of it coming from the user). The problem randomly appeared around 3 weeks ago and consist in an image created using `matplotlib.pyplot.grid`. The image is successfully saved and it also appears in the DOCX file **_if no other images are used in the body of the docx_**. I did some research about this and  I will let here the findings, might be helpful:

First, the problem is not present on all versions of Office. I will underline below the version I tried on (❌ for doesn't work, ✔️ for it works)

-  For Office 2019 version MSO (16.0.14228.20200) 64-bit❌
-  For Office 2019 version MSO (16.0.11901.20170) 64-bit✔️

**Comment by srutherfordbta** [comment]
@dragosnsandu Thanks for nailing down our suspicions around a new version of Windows Word being the catalyst for the break. Do you have images in the header and/or footer as well? For the record, I am generating my images with graphviz as png files, but I don't think the source of the image is a concern at this point.

**Comment by dragosnsandu** [comment]
@srutherfordbta Yes, I confirm that we also have images in the header section of the document. The image from the header is not programatically added, but it is included via the DOCX template I am using.

Another detail, the issues is not related to what version of the `docxtpl` package we are using (as the same behavior applies for both version 0.8.1 and also the most recent one, 0.11.5)

**Comment by srutherfordbta** [comment]
That makes sense on the docxtpl version. The answer to "What has changed?" is the Windows version of Word. It is hard to say if it is a bug in the Windows Word version or if it is something intentionally deprecated/dropped, with Office 365 and the Mac versions getting the same code update down the road. I don't think any of us have insight to that. My fear at this point is that Office 365 version gets updated with the same code and we lose the workaround of opening the document in Office 365, making a small change, saving, downloading and opening correctly in Windows Word. I am really hoping Eric can do some wizardry to get docxtpl to generate the same structure that Office 365 creates once he gets his version of Windows Word.

**Comment by srutherfordbta** [comment]
I just had a look at the docx repo issues and it looks like they are seeing the same thing in the issue below. Let's keep an eye on their issue to see if they come to any kind of resolution.
https://github.com/python-openxml/python-docx/issues/981#issuecomment-891395503

**Comment by elapouya** [comment]
I finally succeeded to reproduce the problem with MS Word Version 2107 on Windows 10.
Now I can investigate...

**Comment by elapouya** [comment]
After building 30+ docx by hand with zip, I found the reason : 
There is a collision with an ID between the header images and the body : 
```
<wp:docPr id="1" name="Picture 1"/>
```
Is the same for header picture and first picture in the body. If I put manually a random ID number for the image in header, all images are now displayed in the body.

I have no idea yet how python-docx manage this ID : I investigate...

**Comment by elapouya** [comment]
Please could you try docxtpl 0.12.0 I just released : it should solve the problem.

**Comment by cw0516** [comment]
Ummmm... unfortunately I could not make it...😥
![version](https://user-images.githubusercontent.com/48001039/129481696-a103d486-9858-4f56-ba2e-e93c9e5b7482.JPG)
![word_version](https://user-images.githubusercontent.com/48001039/129481705-f410d50a-c3a4-4bb6-af8d-7e63b6fdb126.JPG)
![output](https://user-images.githubusercontent.com/48001039/129481709-9be7035b-8b59-4810-ab8d-703497580137.JPG)

**Comment by cw0516** [comment]
wow..................it works!!!!!!!!!!!!!!!!!!!!!!! Thank you Eric !!! it's amazing.........🎉🎉🎉🎉

**Comment by srutherfordbta** [comment]
I have confirmed this fixed my production code and not just the example code. Great work Eric!

**Comment by dragosnsandu** [comment]
Sorry for the delay, just wanted to also confirm that the problem is solved with the latest release. Thank you for your effort, Eric, awesome work!

**Comment by awtimmering** [comment]
I just ran into this problem for the first time today, also confirming a bump to 0.12.0 fixed it. Thanks so much!
